test/profiler/test_profiler.py

```diff
@@ -1505,6 +1505,23 @@ def test_allocation_ids_with_other_ops(self) -> None:
             lambda: torch.zeros((1,)).cos()
         )
 
+    def test_impl_reuse(self) -> None:
+        repeats = 1_000
+        with profile(profile_memory=True, record_shapes=True) as p:
+            for _ in range(repeats):
+                torch.ones((1,))
+                gc.collect()
```

Comment:
This shouldn't be needed? And will slow down this loop quite significantly.

Reply:
I've been pretty liberal with `gc.collect()` to reduce flakiness since my understanding is CPython gets to decide when objects are actually destroyed. (At least that's been my experience putting cleanup code in `__del__`) Or is that only cycle breaking GC?

Reply:
Although as I look at it there's no need for it to be in the inner loop, and 1000 gc collect calls is not very neighborly of me...

Reply:
CPython will actually deallocate your object as soon as the refcount goes to 0.
If there is a cycle involved, the refcount will never get to 0 and so yes, you'll have to wait for the gc to kick in.

So unless you know you're creating cycles, you don't need gc collect in general.